web/src/components/base/Snackbar.tsx

```diff
@@ -0,0 +1,103 @@
+import CloseIcon from '@mui/icons-material/Close'
+import Alert, { alertClasses } from '@mui/material/Alert'
+import AlertTitle from '@mui/material/AlertTitle'
+import Button from '@mui/material/Button'
+import IconButton from '@mui/material/IconButton'
+import MUISnackbar from '@mui/material/Snackbar'
+import { styled } from '@mui/material/styles'
+import React, { ReactElement } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link } from 'react-router-dom'
+
+import { ExtendedSendingStatusType, SNACKBAR_AUTO_HIDE_DURATION } from 'shared'
+
+const StyledMuiSnackbar = styled(MUISnackbar)`
+  position: fixed;
+  bottom: ${props =>
+    props.theme.dimensions.bottomNavigationHeight ?? props.theme.dimensions.visibleFooterHeight}px !important;
+  inset-inline-end: 16px;
+  margin-bottom: 16px;
+`
+
+const StyledAlert = styled(Alert)`
+  display: flex;
+  align-items: center;
+
+  .${alertClasses.action} {
+    padding: 0 8px;
+  }
+`
+
+type SnackbarCloseHandler = (
+  event?: React.SyntheticEvent | Event,
+  reason?: 'timeout' | 'clickaway' | 'escapeKeyDown',
+) => void
+
+export const handleClose =
+  (setOpen: (open: boolean) => void): SnackbarCloseHandler =>
+  (event?, reason?) => {
+    if (reason === 'clickaway') {
+      return
+    }
+    setOpen(false)
+  }
+
+export type SnackbarProps = {
+  open: boolean
+  onClose: SnackbarCloseHandler
+  sendingStatus: ExtendedSendingStatusType
+  dashboardRoute?: string
+  successMessage?: string
+  errorMessage?: string
+  errorTitle?: string
+  autoHideOnSuccess?: boolean
+}
+
+const Snackbar = ({
+  open,
+  onClose,
+  sendingStatus,
+  dashboardRoute,
+  autoHideOnSuccess = false,
```

Comment:
🔧 If you use a severity below, I think you can omit this prop and set the correct hide duration directly.